examples/api/lib/material/toggle_buttons/toggle_buttons.0.dart

```diff
@@ -0,0 +1,158 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for ToggleButtons
+
+import 'package:flutter/material.dart';
+
+const List<Widget> fruits = <Widget>[
+  Text('Apple'),
+  Text('Banana'),
+  Text('Orange')
+];
+
+const List<Widget> vegetables = <Widget>[
+  Text('Tomatoes'),
+  Text('Potatoes'),
+  Text('Carrots')
+];
+
+const List<Widget> icons = <Widget>[
+  Icon(Icons.sunny),
+  Icon(Icons.cloud),
+  Icon(Icons.ac_unit),
+];
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  static const String _title = 'ToggleButtons Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: _title,
+      home: ToggleButtonsSample(title: _title),
+    );
+  }
+}
+
+class ToggleButtonsSample extends StatefulWidget {
+  const ToggleButtonsSample({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  State<ToggleButtonsSample> createState() => _ToggleButtonsSampleState();
+}
+
+class _ToggleButtonsSampleState extends State<ToggleButtonsSample> {
+  final List<bool> _selectedFruits = <bool>[true, false, false];
+  final List<bool> _selectedVegetables = <bool>[false, true, false];
+  final List<bool> _selectedWeather = <bool>[false, false, true];
+  bool vertical = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+
+    return Scaffold(
+      appBar: AppBar(title: Text(widget.title)),
+      body: Center(
+        child: SingleChildScrollView(
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              // ToggleButtons with a single selection.
+              Text('Single-select', style: theme.textTheme.subtitle2),
+              const SizedBox(height: 5),
+              ToggleButtons(
+                direction: vertical ? Axis.vertical : Axis.horizontal,
+                onPressed: (int index) {
+                  setState(() {
+                    // The button that is tapped is set to true, and the others to false.
+                    for (int i = 0; i < _selectedFruits.length; i++) {
+                      _selectedFruits[i] = i == index;
+                    }
```

Comment:
I mean, that is simpler, which is good, but I was thinking of `_selectedFruits` being an int containing the selected fruit, which is even simpler:

```suggestion
                    _selectedFruit = index;
```

Reply:
ToggleButton expects a list of bools `required List<bool> isSelected,` to get the selected items

Reply:
This bool list is checked for index to select or unselect, if `_selectedFruits` is just int then It would have to be converted to a bool list for `isSelected` parameter